schemas/Identity.json

```diff
@@ -0,0 +1,22 @@
+{
+    "type": "object",
+    "description": "The user properties to apply when identifying",
+    "properties": {
+      "eventName": {
+        "enum": ["Identity"]
+      },
+      "ftueUseCaseSelection": {
+        "description": "The selected messaging use case during the onboarding flow.",
+        "type": "string",
+        "enum": [
+          "FriendsAndFamily",
+          "Teams",
+          "Communities",
+          "Skipped"
+        ]
+      }
+    },
+    "required": [],
```

Comment:
should the properties be enforced? Clients opting in vs opting out

Reply:
I would say it's fine to leave `ftueUseCaseSelection` as optional. Potentially for this one event, we might want to leave all the properties as optional so we could use `$set` from elsewhere in the app on a property by property basis. WDYT @novocaine?

Reply:
I think that makes sense with the caveat that clients will need to have a $set wrapper that still does the validation

Reply:
That sounds sensible to me. If I've interpreted the docs correctly, I think the right option here would be to have a `setUserProperties(from identity: IdentityAnalyticsEvent` method that caches the Identity event and attaches it to the next event we send.

Reply:
I've been thinking about this a bit more, caching the identity event would also mean serialising the event to prevent data loss, which I'd prefer to avoid if possible!

from the docs I couldn't find any mention of avoiding calling identify directly, we could have a mix of 

```kotlin
fun updateUserProperties(identity: IdentityEvent) {
  // calls identify with the currently known analytics id and identity payload
}

fun capture(
  event: Event, 
  setIdentity: IdentityEvent? = null, 
  setOnceIdentity: IdentityEvent? = null
) {
  // applies optional $set and $set_once properties if present to the event payload
}
```

what do you think?

Reply:
> caching the identity event would also mean serialising the event to prevent data loss, which I'd prefer to avoid if possible!

Should we have a way to serialise the identity anyway? For example - I sign up and choose not to opt in when prompted for analytics. Later on I find the toggle in settings and decide to opt in. It would be a shame to lose any potential user properties in this instance (although this is probably more of an edge case in reality).

Maybe we should discuss. Although ultimately it's no more than an implementation detail specific to the client. So long as we're sticking to the schema, it probably doesn't matter that much.